src/main/java/com/entropyteam/entropay/employees/services/PtoService.java

```diff
@@ -271,4 +264,20 @@ private Double getPtoHours(Pto pto, LocalDate startDate, LocalDate endDate,
                 .filter(date -> !holidayDates.contains(date)) // Exclude holidays
                 .count() * 8.0;
     }
+
+    public List<PtoTimeEntry> findPtoActivities(LocalDate startDate, LocalDate endDate) {
+        Map<Country, Set<LocalDate>> holidaysByCountry = holidayService.getHolidaysByCountry(startDate, endDate);
+
+        List<PtoTimeEntry> ptoActivities = new ArrayList<>();
+
+        ptoRepository.findAllBetweenPeriod(startDate, endDate).forEach(
+                pto -> pto.getStartDate().datesUntil(pto.getEndDate().plusDays(1))
+                        .filter(date -> date.getDayOfWeek().getValue() < 6) // Include only weekdays
```

Comment:
Minor: We are using 2 different ways to check if day is in weekend or not, here and in L178. Would be nice to use the same for both and maybe extract to a separate method.